File: advent_of_code/5/advent_5_java/src/main/java/com/Move.java
package com;

public record Move(int quantity, int from, int to) {

  public static Move from(String line) {
    var split = line.split(" ");
    return new Move(Integer.parseInt(split[1]), Integer.parseInt(split[3]) - 1, Integer.parseInt(split[5]) - 1);
  }
}

File: advent_of_code/5/advent_5_node/Move.js
export default class Move {
  constructor(quantity, from, to) {
    this.quantity = quantity;
    this.from = from;
    this.to = to;
  }

  static from(line) {
    const split = line.split(" ");
    return new Move(Number(split[1]), Number(split[3]) - 1, Number(split[5]) - 1);
  }
}
